chrome/model.js

```diff
@@ -223,6 +219,79 @@ GmailToTrello.Model.prototype.loadTrelloMembers = function(boardId) {
     });
 };
 
+GmailToTrello.Model.prototype.loadThreadTrelloCards = function () {
+    var self = this;
+
+    self.isThreadCardsLoaded = true;
+
+    // get cards associated with the thread on the current thread
+    // i would love to do this with the message ids, but there's no way to get all of them to search
+    // so let's use the subject and time method
+    if (!self.gmail || !self.gmail.subject || self.gmail.subject.length < 1) {
+        gtt_log('loadThreadTrelloCards: Current page is not a thread');
+        self.trello.threadCards = [];
+        self.event.fire('onLoadThreadTrelloCardsSuccess');
+    }
+    else {
+        gtt_log('loadThreadTrelloCards: Cards for current thread');
+        var query = encodeURIComponent(self.gmail.subject);
+        Trello.get('search', {'query': query, 'modelTypes': 'cards'}, function(data) {
```

Comment:
This uses Trello's "search" method to find cards associated with the current thread across all of the member's boards in one request. I originally pulled the ID out of the URL and searched for the ID, which was the obvious approach. But the ID in the URL is actually a message ID, not a thread ID, so you have a bunch of different message IDs in the same thread, meaning the ID in the Trello card may not match the ID when you access the page.

So my second attempt was to find somewhere in Gmail where I could get all of the message IDs in the current thread. But this is quite a bit easier said than done. I did actually see a list of them in the heap snapshot, but I couldn't find a way to pull them out of the page or request them from Gmail or anything without getting Gmail API access, which seemed like overkill.

So instead, I'm searching for the urlencoded subject line, which appears in the "Time" link on each card. Then I'm pulling the time from that same URL on the Trello card and searching within 1d of that time. If it matches the current thread, we add the card to threadTrelloCards.